pkg/kubelet/prober/worker.go

```diff
@@ -316,11 +316,14 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 
 	w.resultsManager.Set(w.containerID, result, w.pod)
 
-	if (w.probeType == liveness || w.probeType == startup) && result == results.Failure {
+	if (w.probeType == liveness && result == results.Failure) || (w.probeType == startup && (result == results.Success || result == results.Failure)) {
```

Comment:
```suggestion
	if (w.probeType == liveness && result == results.Failure) || (w.probeType == startup && result != results.Unknown) {
```

Reply:
@HirazawaUi 
Thank you for your comment.

> After the w.probeManager.prober.probe() is executed, will the startup probe still enter the unknown state?

As I've checked the codebase, `w.probeManager.prober.probe()` will return err with `unknown` state.
So I think it's enough to check `w.probeType == startup`.

> As long as the startup probe exceeds its FailureThreshold or SuccessThreshold, regardless of whether it succeeds or fails, we will stop executing it. Is that correct?

As far as I can tell from checking the [doc](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes), that seems correct.
Do you have any thoughts on this?

Reply:
It looks good to me now.
/lgtm